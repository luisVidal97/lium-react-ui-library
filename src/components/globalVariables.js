
const forms = ['contained', 'outlined'];

const heights = {
  large: '50px',
  medium: '40px',
  small: '30px',
}

const borderRadius = {
  strong: '20px',
  normal: '10px',
  none: '0px'
}

const heightsInputText = {
  large: {
    shape: '40px',
    font: '1.2rem'
  },
  medium:  {
    shape: '32px',
    font: '1rem'
  },
  small: {
    shape: '24px',
    font: '.8rem'
  },
}


const colors = {
  primary: {
      normal: '#115293',
      hover: '#0A0A36',
  },
  secondary: {
    normal: '#DC004E',
    hover: '#9A0036',
  },
  danger: {
      normal: '#ff0000',
      hover: '#cc0000',
  },
  warning: {
    normal: '#ffea05',
    hover: '#e2d006'
  },
  success: {
    normal: '#06c920',
    hover: '#019e16'
  },
  dark: {
    normal: '#000000',
    hover: '#686767',
  },
};

const defaultStyle = {
  normal: '#000000',
  hover: '#686767',
}

export { forms, borderRadius, heights, heightsInputText, colors, defaultStyle } ;
