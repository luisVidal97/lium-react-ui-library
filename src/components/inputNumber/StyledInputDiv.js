import styled from 'styled-components';

export const StyledInputDiv = styled.div`

  display: inline-block;
  position: relative;
  color: #A6A6A6;
  &::after {
    position: absolute;
    top: 25%;
    left: .5em;
  }
  &::after{
    content: '${props => props.prefix}';
    font-size: ${props => props.fontSize};
  }
`;
