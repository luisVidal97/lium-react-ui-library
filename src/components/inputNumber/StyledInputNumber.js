import styled from 'styled-components';

export const StyledInputNumber = styled.input`
  margin: 5px;
  height: ${ props => props.shapeSize};
  font-size: ${ props => props.fontSize};
  border-radius: ${ props => props.border};
  border: 1px solid #ccc;
  background-color: ${ props => props.backgroundColor};
  pointer-events: ${ props => props.pointerEvent};

  &:hover{
    border-color: ${props => props.color};
  }

  &:focus{
    outline: none !important;
    border-color: ${props => props.color};
    box-shadow: 0 0 5px ${props => props.color};
  }


  padding-left: 20px;
`;

export const StyledInputDiv = styled.div`

  display: inline-block;
  position: relative;
  color: #A6A6A6;
  &::after {
    position: absolute;
    top: 25%;
    left: .5em;
  }
  &::after{
    content: '${props => props.prefix}';
    font-size: ${props => props.fontSize};
  }
`;
