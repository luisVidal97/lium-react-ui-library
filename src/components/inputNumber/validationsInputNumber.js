import {colors, heightsInputText, borderRadius} from '../globalVariables'


export const validationsInputNumber = ( { prefix, disabled: disabledInputText, color, size, border} ) =>{

  prefix = prefix?
    prefix.charAt(0)
    : '';

  const shapeSize = size && heightsInputText[size]?
  heightsInputText[size].shape
  : heightsInputText['medium'].shape;

  const fontSize = size && heightsInputText[size]?
  heightsInputText[size].font
  : heightsInputText['medium'].font;

  color = color && colors[color]?
    colors[color].normal
    :
    colors['primary'].normal;

  border = border && borderRadius[border]?
    borderRadius[border]
    :
    borderRadius['normal'];

  let backgroundColor = 'transparent', pointerEvent = 'auto';
  if (disabledInputText === 'true') {
    backgroundColor = '#E0E0E0' ;
    pointerEvent = 'none'
    color = 'transparent';
  }

  return {
    prefix,
    shapeSize,
    fontSize,
    color,
    border,
    backgroundColor,
    pointerEvent,
  };
}

