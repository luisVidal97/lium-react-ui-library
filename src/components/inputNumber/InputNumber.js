import React  from 'react';
import PropTypes from 'prop-types';
import { StyledInputNumber } from './StyledInputNumber';
import { StyledInputDiv } from './StyledInputDiv';
import { validationsInputNumber } from './validationsInputNumber';

const InputNumber = (props) => {


  const sources = validationsInputNumber(props);

  const validateNumbers = (e) => {
    e.target.value = e.target.value.replace(/[^0-9.]/g, '');

    if (props.formatter === "true"){

      if( e.target.value.includes('.')){
        const array = e.target.value.split('.');
        e.target.value =array[0].replace(/(,*)/g, '').replace(/\B(?=(\d{3})+(?!\d))/g, ',') +'.'+ array[1];

      }else {
        e.target.value = e.target.value.replace(/(,*)/g, '').replace(/\B(?=(\d{3})+(?!\d))/g, ',')
      }
    }
  }

  return (
    <StyledInputDiv fontSize={sources.fontSize} prefix={sources.prefix} >
      {
        props.onChange?
          <StyledInputNumber
            type="text"
            maxLength={props.maxLength}
            placeholder={props.placeholder}
            name={props.name}
            onInput = { validateNumbers }
            onChange = {(e)=>{
              props.onChange(e, e.target.value.replace(/(,*)/g, ''));
            }}
            {...sources}
          />
        :
          <StyledInputNumber
            type="text"
            maxLength={props.maxLength}
            placeholder={props.placeholder}
            name={props.name}
            onInput = { validateNumbers }
            {...sources}
          />
      }

    </StyledInputDiv>
  )
}

InputNumber.propTypes = {
  prefix: PropTypes.string,
  border: PropTypes.string,
  size: PropTypes.string,
  disabled: PropTypes.string,
  text: PropTypes.string,
  color: PropTypes.string,
  placeholder: PropTypes.string,
  maxLength: PropTypes.string,
  name: PropTypes.string,
  onChange: PropTypes.func,
}

export default InputNumber;
