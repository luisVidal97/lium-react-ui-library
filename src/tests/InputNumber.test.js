import React from 'react';
import { mount, shallow } from'enzyme';

import 'jest-styled-components';
import renderer from 'react-test-renderer';

import { heightComponents, colors, borderRadius } from '../components/globalVariables';
import InputNumber from '../components/inputNumber/InputNumber';
import {StyledInputNumber} from '../components/inputNumber/StyledInputNumber';

describe('Tests over <InputNumber/>: features and behavior', () => {

  let wrapper, tree;

  test('should show <Button/> successfully', () => {
    wrapper = shallow(<InputNumber />);
    expect( wrapper ).toMatchSnapshot();
  });

  test(`should show default style values of text input invoking <InputNumber /> component (without props).
  Default values = height: 32px; font-size: 1rem; border-radius: 10px; background-color: tranparent; pointer-events: auto;
  input[focus]border-color: #115293;  input[focus]box-shadow
  : 0 0 5px #115293;
  `, () => {
    tree = renderer.create(<InputNumber />).toJSON();
    const child = tree.children[0];
    expect(child).toHaveStyleRule('height', heightComponents.medium.shape);
    expect(child).toHaveStyleRule('font-size', heightComponents.medium.font);
    expect(child).toHaveStyleRule('border-radius', borderRadius.normal);
    expect(child).toHaveStyleRule('background-color', 'transparent');
    expect(child).toHaveStyleRule('pointer-events', 'auto');
    expect(child).toHaveStyleRule('border-color', colors.primary.normal, { modifier: ':focus', });
    expect(child).toHaveStyleRule('box-shadow', `0 0 5px ${colors.primary.normal}`, { modifier: ':focus', });
  });

  test('should show default style values of text input invoking <InputNumber /> (with props but wrong values)', () => {
    tree = renderer.create(<InputNumber disabled='treu' size='sm' color='green' border='24' moove='large'/>).toJSON();
    const child = tree.children[0];
    expect(child).toHaveStyleRule('height', heightComponents.medium.shape);
    expect(child).toHaveStyleRule('font-size', heightComponents.medium.font);
    expect(child).toHaveStyleRule('border-radius', borderRadius.normal);
    expect(child).toHaveStyleRule('background-color', 'transparent');
    expect(child).toHaveStyleRule('pointer-events', 'auto');
    expect(child).toHaveStyleRule('border-color', colors.primary.normal, { modifier: ':focus', });
    expect(child).toHaveStyleRule('box-shadow', `0 0 5px ${colors.primary.normal}`, { modifier: ':focus', });
  });


  test('should show secondary, large and with strong border input text', () => {
    tree = renderer.create(<InputNumber prefix="$" color='secondary' size='large' border='strong'/>).toJSON();
    const child = tree.children[0];
    expect(child).toHaveStyleRule('pointer-events', 'auto');
    expect(child).toHaveStyleRule('border-color', colors.secondary.normal, { modifier: ':focus', });
    expect(child).toHaveStyleRule('box-shadow', `0 0 5px ${colors.secondary.normal}`, { modifier: ':focus', });
    expect(child).toHaveStyleRule('height', heightComponents.large.shape);
    expect(child).toHaveStyleRule('font-size', heightComponents.large.font);
    expect(child).toHaveStyleRule('border-radius', borderRadius.strong);
    expect(child).toHaveStyleRule('pointer-events', 'auto');
  });


  test('should show disabled input text', () => {
    tree = renderer.create(<InputNumber disabled="true" size='small' border="none"/>).toJSON();
    const child = tree.children[0];
    expect(child).toHaveStyleRule('pointer-events', 'none');
    expect(child).toHaveStyleRule('background-color', '#E0E0E0');
    expect(child).toHaveStyleRule('border-radius', borderRadius.none);
    expect(child).toHaveStyleRule('border-color', 'transparent', { modifier: ':hover', });
    expect(child).toHaveStyleRule('border-color', 'transparent', { modifier: ':focus', });
    expect(child).toHaveStyleRule('box-shadow', `0 0 5px transparent`, { modifier: ':focus', });
  });

  test('should not execute function if input text is disabled', () => {
    const doSomething = jest.fn();
    wrapper = shallow( <InputNumber disabled="true" placeholder="Hello world!" value="Hello" onChange={doSomething}/> );
    wrapper.find(StyledInputNumber).simulate('click');
    expect(doSomething).not.toHaveBeenCalled();
  });


  test('should change the value of the const if input text change', () => {
    const doSomething = jest.fn();
    const ph = 'number';
    wrapper = shallow( <InputNumber placeholder={ph} onChange={doSomething}/> );
    wrapper.find(StyledInputNumber).simulate('change', { target: { value:'123' }});
    expect(doSomething).toHaveBeenCalled();
    expect( wrapper.find(StyledInputNumber).props().placeholder).toBe(ph);
  });


  test('should not allow a letter value(removing it of string) or remove comma if it does not  formatter prop like true', () => {
    const doSomething = jest.fn();
    wrapper = shallow( <InputNumber value="25,25" onChange={doSomething}/> );
    const input = wrapper.find(StyledInputNumber);
    expect(input.props().value).toBe('2525');

    wrapper = shallow( <InputNumber value="25a25" onChange={doSomething}/> );
    const input2 = wrapper.find(StyledInputNumber);
    expect(input2.props().value).toBe('2525');

  });


  test('should show the formatted value in the box', () => {
    const doSomething = jest.fn();
    const number = '12345678';
    wrapper = shallow( <InputNumber formatter="true" value={number} onChange={doSomething}/> );
    const input = wrapper.find(StyledInputNumber);
    expect(input.props().value).toBe('12,345,678');

    const number2 = '1234567.8';
    wrapper = shallow( <InputNumber formatter="true" value={number2} onChange={doSomething}/> );
    const input2 = wrapper.find(StyledInputNumber);
    expect(input2.props().value).toBe('1,234,567.8');

  });

});
