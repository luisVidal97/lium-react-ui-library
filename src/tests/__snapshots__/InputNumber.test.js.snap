// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests over <InputNumber/>: features and behavior should show <Button/> successfully 1`] = `
<styled.div
  fontSize="1rem"
  prefix=""
>
  <styled.input
    backgroundColor="transparent"
    border="10px"
    color="#115293"
    fontSize="1rem"
    onInput={[Function]}
    pointerEvent="auto"
    prefix=""
    shapeSize="32px"
    type="text"
  />
</styled.div>
`;
